jsapp/js/components/special/koboSelect3.tsx

```diff
@@ -0,0 +1,391 @@
+
+      // Select first option (if clearable, clears the selection)
+      case 'PageUp':
+      case 'Home':
+        indexRef.current = homeIndex();
+        optionRef.current = optionAt(indexRef.current);
+        e.preventDefault();
+        break;
+
+      // Select last option
+      case 'PageDown':
+      case 'End':
+        indexRef.current = endIndex();
+        optionRef.current = optionAt(indexRef.current);
+        e.preventDefault();
+        break;
+    }
+
+    // Letter keystroke matching lives in a helper function
+    emulateBrowserSelectLetterMatching(e.key);
+
+    // Menu open/close affects what the Tab/Enter/Space/Esc keys do,
+    // and whether the selection "commits" (sends a change event) or not
+    if (expanded) {
+      scrollOptionIntoView();
+      switch (e.key) {
+        case 'Tab':
+        case 'Enter':
+          commitOption();
+          closeMenu();
+          break;
+        case 'Escape':
+          resetRefs(); // Revert selection to props value
+          closeMenu();
+          break;
+      }
+    } else {
+      // If the menu is closed, immediately commit selection changes.
+      commitOption();
+      switch (e.key) {
+        case ' ':
+        case 'Enter':
+          openMenu();
+          // scrollTriggerIntoView();
+          break;
+      }
+    }
+
+    // Most of the code paths above change indexRef and optionRef.
+    // We'd like to trigger a re-render.
+    forceUpdate();
+  };
+
+  const triggerBlurHandler = () => {closeMenu();};
+  const triggerMouseDownHandler = (e: React.MouseEvent) => {
+    if (e.button === 0) {
+      toggleMenu();
+    }
+  };
+
+  // Mousedown.
+  const optionMouseHandler = (e: React.MouseEvent) => {
+    const optionValue = (e.target as HTMLElement).dataset?.value;
+    // If that's a valid option, commit it!
+    const index = props.options.findIndex((o) => o.value === optionValue);
+    if (index > -1) {
+      indexRef.current = index;
+      optionRef.current = optionAt(index);
+      // Commit if it's a left or middle click!
+      if (e.button < 2 && (e.type === 'mouseup' || e.type === 'mousedown')) {
+        commitOption();
+        closeMenu();
+        setTimeout(() => triggerElRef.current?.focus(), 0);
+      }
+    }
+    forceUpdate();
+  };
+  const preventDefault = (e: React.UIEvent) => {e.preventDefault();};
+
+  return (
+    <div className={styles.root}>
+      <label className={styles.label} htmlFor={'select-' + props.name}>
+        {props.label}
+        {props.required && <span className={styles.redAsterisk}> * </span>}
+      </label>
+      {/* Trigger */}
+      <button
+        className={cx(
+          styles.trigger,
+          props.value || styles.placeholding,
+          props.error && styles.hasError,
+        )}
+        type='button'
+        // ARIA. There's room for improvement here.
+        // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role
+        aria-role='combobox'
```

Comment:
It looks this this should be `role` instead of `aria-role` - it fixes this error in the console:

![image](https://github.com/kobotoolbox/kpi/assets/7819986/f4ee7330-4df4-49a1-a05f-d65dd1302a36)

(Which is delightfully inconsistent on React's part.)